x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/hooks/use_investigation_fields.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useMemo } from 'react';
+import type { Datatable, ExpressionsStart } from '@kbn/expressions-plugin/public';
+import type { DataViewFieldBase } from '@kbn/es-query';
+import { computeIsESQLQueryAggregating } from '@kbn/securitysolution-utils';
+
+import { useQuery } from '@tanstack/react-query';
+
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+
+import { getEsqlQueryConfig } from '../../rule_creation/logic/get_esql_query_config';
+
+const esqlToFields = (
+  data: { error: unknown } | Datatable | undefined | null
+): DataViewFieldBase[] => {
+  if (data && 'error' in data) {
+    return [];
+  }
+
+  const fields = (data?.columns ?? []).map(({ id, meta }) => {
+    return {
+      name: id,
+      type: meta.type,
+    };
+  });
+
+  return fields;
+};
+
+type UseEsqlFields = (esqlQuery: string | undefined) => {
+  isLoading: boolean;
+  fields: DataViewFieldBase[];
+};
+
+/**
+ * fetches ES|QL fields and convert them to DataViewBase fields
+ */
+const useEsqlFields: UseEsqlFields = (esqlQuery) => {
+  const kibana = useKibana<{ expressions: ExpressionsStart }>();
+
+  const { expressions } = kibana.services;
+
+  const queryConfig = getEsqlQueryConfig({ esqlQuery, expressions });
+  const { data, isLoading } = useQuery(queryConfig);
+
+  const fields = useMemo(() => {
+    return esqlToFields(data);
+  }, [data]);
+
+  return {
+    fields,
+    isLoading,
+  };
+};
+
+type UseInvestigationFields = (params: {
+  isEsqlRule: boolean;
+  esqlQuery: string | undefined;
+  indexPatternsFields: DataViewFieldBase[];
+}) => {
+  isLoading: boolean;
+  investigationFields: DataViewFieldBase[];
+};
+
+export const useInvestigationFields: UseInvestigationFields = ({
+  isEsqlRule,
```

Comment:
Do we need `isEsqlRule` parameter? I guess if rule is not ESQL then `esqlQuery` will be `undefined` and that should cover it

Reply:
agree, can be simplified, removed it